src/main/java/com/google/cloud/genomics/dataflow/functions/AlleleSimilarityCalculator.java

```diff
@@ -58,38 +58,39 @@ public void processElement(ProcessContext context) {
     CallSimilarityCalculator callSimilarityCalculator =
         callSimilarityCalculatorFactory.get(isReferenceMajor(variant));
     FluentIterable<KV<Call, Call>> pairs =
-        PairGenerator.<Call, ImmutableList<Call>>allPairs(getSamplesWithVariant(variant));
+        PairGenerator.<Call, ImmutableList<Call>>allPairs(getSamplesWithVariant(variant), false,
+            new VariantUtils.CallComparator());
     for (KV<Call, Call> pair : pairs) {
       accumulateCallSimilarity(callSimilarityCalculator, pair.getKey(), pair.getValue());
     }
-    ++numberOfVariants;
   }
 
   private void accumulateCallSimilarity(CallSimilarityCalculator callSimilarityCalculator,
       Call call1, Call call2) {
     KV<String, String> callPair = KV.of(call1.getCallSetName(), call2.getCallSetName());
-    if (!accumulator.containsKey(callPair)) {
-      accumulator.put(callPair, 0.0);
+    KV<Double, Integer> callPairAccumulation = accumulator.get(callPair);
+    if (callPairAccumulation == null) {
+      callPairAccumulation = KV.of(0.0, 0);
+      accumulator.put(callPair, callPairAccumulation);
     }
-    accumulator.put(callPair,
-        accumulator.get(callPair) + callSimilarityCalculator.similarity(call1, call2));
+    accumulator.put(callPair, KV.of(
+        callPairAccumulation.getKey() + callSimilarityCalculator.similarity(call1, call2),
+        callPairAccumulation.getValue() + 1));
```

Comment:
the way that you are incrementing the values of the KV....it reminds me strongly of guava's Multiset. Might be worth investigating using it instead - the downside is that you'd have to write a Coder for it.